File: Java/Binary-Search/binarySearch.java

public class binarySearch {

    public static void main(String[] args) {
        int[] arr = {11,23,45,67,78,89,90};
        int target = 7;
        System.out.println(search(arr, target);
    }


    static void search(int [] arr, target){
        int s = 0;
        int e = arr.length -1 ;
        while(s <= e)
        {

            if(target < arr[0]){
                return -1;
            }
            int mid = s + (e-s)/2;
            if(target == arr[mid])
            {
                return mid;
            }
            else if(target > arr[mid])
            {
                s = mid+1;
            }

            else {
                e = mid-1;
            }
        }
        return mid;
    }


    //CEILING AND FLOOR
    static int floor(int[] arr, int target)
    {
        int s = 0;
        int e = arr.length -1 ;
        while(s <= e)
        {

            if(target < arr[0]){
                return -1;
            }
            int mid = s + (e-s)/2;
            if(target == arr[mid])
            {
                return mid;
            }
//            if (isAsc){
                else if(target > arr[mid])
                {
                    s = mid+1;
                }

                else {
                    e = mid-1;
                }
//            }

//            else {
//                if(target < arr[mid])
//                {
//                    s = mid+1;
//                }
//
//                else if(target > arr[mid]){
//                    e = mid-1;
//                }
//            }
        }
        //        flooring
        return arr[e];

    }
}

File: C/Arrays/binary_search.c
#include <stdio.h>

int binarySearch(int arr[], int size, int key)
{

    int start = 0, end = size - 1;

    while (start <= end)
    {

        int mid = start + (end - start) / 2;

        if (arr[mid] == key)
        {
            return mid;
        }

        else if (arr[mid] > key)
        {
            end = mid - 1;
        }

        else
        {
            start = mid + 1;
        }
    }

    // if element is not found in array
    return -1;
}

int main()
{

    // let's take input from user of hw many elements he/she wants to enter in our array
    int size;
    printf("Enter the number of elements you want to enter in array: ");
    scanf("%d", &size);

    int arr[size];

    printf("\nPlease, start entering the elements in array: \n");
    for (int i = 0; i < size; i++)
    {
        printf("Enter element at %d index of array: ", i);
        scanf("%d", &arr[i]);
    }

    int key;
    printf("\nEnter the element you want to search in the array: ");
    scanf("%d", &key);

    int found = binarySearch(arr, size, key);

    if (found == -1)
    {
        printf("\nElement is not present in the inputted array");
    }
    else
    {
        printf("\nElement is present at %d index in the array", found);
    }

    return 0;
}